packages/web-components/fast-foundation/src/design-token/core/design-token-node.ts

```diff
@@ -0,0 +1,694 @@
+                // If this is not the originator, check to see if this node
+                // has any dependencies of the token value. If so, we need to evaluate for this node
+                let evaluate = false;
+
+                for (const dependency of dependencies) {
+                    if (DesignTokenNode.isAssigned(this, dependency)) {
+                        evaluate = true;
+                        break;
+                    }
+                }
+
+                if (evaluate) {
+                    const prev = this._derived.get(token)?.value;
+                    const derivedValue = this.setupDerivedTokenValue(token, value);
+
+                    if (prev !== derivedValue.value) {
+                        const type =
+                            prev === undefined
+                                ? DesignTokenMutationType.add
+                                : DesignTokenMutationType.change;
+                        const notification = new DesignTokenChangeRecordImpl(
+                            this,
+                            type,
+                            token,
+                            derivedValue.evaluator.value
+                        );
+                        DesignTokenNode.queueNotification(notification);
+                        record = notification;
+                    }
+                }
+            }
+        }
+
+        this.collectLocalChangeRecords(record).forEach(_record => {
+            DesignTokenNode.queueNotification(_record);
+            this.dispatch(_record);
+        });
+
+        this.notifyChildren(record);
+    }
+
+    /**
+     * Generate change-records for local dependencies of a change record
+     */
+    private collectLocalChangeRecords<T>(
+        record: DesignTokenChangeRecordImpl<T>
+    ): Map<DesignToken<any>, DesignTokenChangeRecordImpl<any>> {
+        const collected = new Map<DesignToken<any>, DesignTokenChangeRecordImpl<any>>();
+        for (const dependent of DesignTokenNode.getOrCreateDependencyGraph(
+            this,
+            record.token
+        )) {
+            if (dependent.value !== dependent.update().value) {
+                collected.set(
+                    dependent.token,
+                    new DesignTokenChangeRecordImpl(
+                        this,
+                        DesignTokenMutationType.change,
+                        dependent.token,
+                        dependent.evaluator.value
+                    )
+                );
+            }
+        }
+
+        return collected;
+    }
+
+    /**
+     *
+     * Notify children of changes to the node
+     */
+    private notifyChildren(...records: DesignTokenChangeRecordImpl<any>[]) {
+        if (this.children.length) {
+            for (let i = 0, l = this.children.length; i < l; i++) {
+                for (let j = 0; j < records.length; j++) {
+                    this.children[i].dispatch(records[j]);
+                }
+            }
+        }
```

Comment:
Not sure if it makes sense, but would there be an efficiencies gained by instead sending the list of records to child instead of using nested for loops to send the individual records? Maybe they just have to loop internally anyway and so there's no advantage. Just thought I'd check.

Reply:
Follow on question, is this the dispatch an array improvement slated for a later date?

Reply:
Yeah I touched on it in our sync, but I think we can see some improvement by reconciling all changes for each node and dispatching a set of mutations instead of dispatching one mutation at a time. I'll file a tracking issue.